lib/bottles.rb

```diff
@@ -1,2 +1,41 @@
 class TenGreenBottles
-end 
+
+
+  def initialize(number)
+    @number = number
+  end
+
+  def number_of_bottles
+    if @number >= 3
+      regular_verse
+    elsif @number == 2
+      singular_verse
+    else
+      final_verse
+    end
+  end
+
+  def regular_verse
+    "#{@number} green bottles hanging on the wall
+#{@number} green bottles hanging on the wall
+And if 1 green bottle should accidentally fall,
+There'll be #{@number - 1} green bottles hanging on the wall."
```

Comment:
It's a super basic calculation, but I'd probably pull the string interpolation out into a method to reduce the number by one.